core/Scheduler/Scheduler.php

```diff
@@ -275,8 +306,15 @@ private function executeTask($task)
             $callable = array($task->getObjectInstance(), $task->getMethodName());
             call_user_func($callable, $task->getMethodParameter());
             $message = $timer->__toString();
-        } catch (Exception $e) {
+        } catch (\Throwable $e) {
+            $this->logger->error("Scheduler: Error {errorMessage} for task '{task}'",
+                ['errorMessage' => $e->getMessage(), 'task' => $task->getName()]);
             $message = 'ERROR: ' . $e->getMessage();
+
+            // If the task has indicated that retrying on exception is safe then flag for rescheduling
+            if ($e->getCode() == 1) {
```

Comment:
This actually doesn't look like a very clean solution. In theory every exception could have the error code 1. So either we use a custom exception (like `RetryableException`) for that, so we can check for that specific class. Or if we want to define that on a task level, dependently from a certain exception, we could have a new property in the Task class `retryCount`, which defaults to `0`, but can be overwritten in the constructor (and the various factories).

Reply:
Agreed, a custom exception is a much better approach, I've committed a change for this. Since it might not be a good idea to retry the task depending on the stage that the task reached when it failed, a custom exception will allow the task code to decide whether a retry should happen or not, which should be more flexible then defining retries at the task class level.